src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -246,15 +246,13 @@ public void AddRange(IEnumerable<T> items)
             {
                 Requires.NotNull(items, "items");
 
-                int count;
-                if (items.TryGetCount(out count))
+                int count = ImmutableExtensions.GetCount(ref items);
```

Comment:
I was just following the same strategy as in InsertRange, but I can see your point because here it's not as necessary to know the size beforehand. I've updated my commit to use the original implementation as a fallback.